src/state/swap/hooks.ts

```diff
@@ -77,7 +78,9 @@ export function tryParseAmount<T extends Currency>(value?: string, currency?: T)
     return undefined
   }
   try {
-    const typedValueParsed = parseUnits(value, currency.decimals).toString()
+    // Here we drop everything after given token decimals to avoid parsing issues
```

Comment:
so this is an issue with the ethers util function?

Reply:
Yeah, it throws if there are more decimals than the token precision to format

Reply:
laaaaame